experiments/example/main.go

```diff
@@ -0,0 +1,129 @@
+	minThreadCountFlag = conf.NewIntFlag("example_min_thread_count", "Minimum count of memcached threads to be tested in the experiment", 1)
+	maxThreadCountFlag = conf.NewIntFlag("example_max_thread_count", "Maximum count of memcached threads to be tested in the experiment", 4)
+)
+
+func main() {
+	// It's always useful to store experiment start time for future reference.
+	experimentStart := time.Now()
+
+	// Configuring application - parsing flags, configuring log level, dumping the configuration (if requested).
+	experiment.Configure()
+
+	// Generate an experiment ID that will be used to tag all the metrics collected as well as to identify experiment metadata.
+	uid := uuid.New()
+
+	// Initialize logger (and log some basic information: experiment name, UUID generated above etc).
+	logger.Initialize(appName, uid)
+
+	// Connect to metadata database (Cassandra is our database of choice).
+	// Besides experiment results and platform metrics (we use Snap to gather them) we save certain deta about experiment configuration and environment (metadata).
+	metadata, err := experiment.NewMetadata(uid, experiment.MetadataConfigFromFlags())
+	// errutil.CheckWithContext() is a helper function that will panic on error and provide some additional information about error origin.
+	errutil.CheckWithContext(err, "Cannot connect to Cassandra Metadata Database")
+
+	// Save experiment runtime environment (configuration, environmental variables, etc).
+	err = metadata.RecordRuntimeEnv(experimentStart)
+	errutil.CheckWithContext(err, "Cannot save runtime environment in Cassandra Metadata Database")
+
+	// We should validate OS configuration to make sure that it is ready to execute the experiment binary.
+	validate.OS()
+
+	// Now we need to get hold of experiment configuration.
+	// Thread count is described at the rop of this file.
+	minThreadCount := minThreadCountFlag.Value()
+	maxThreadCount := maxThreadCountFlag.Value()
+	// Each iteration of memcached stressing will last amount of time that loadDuration defines.
+	loadDuration := sensitivity.LoadDurationFlag.Value()
+	// To determine memcached capacity we will attempt to send various amount of QPS number of times.
+	numberOfLoadPoints := sensitivity.LoadPointsCountFlag.Value()
+	// Maximum amount of QPS to be able to calculate value for each load point.
+	maxQPS := sensitivity.PeakLoadFlag.Value()
+
+	// When configuration flags' values are available we should save it to the metadata database.
+	// Each value needs to be a string so some conversion might be necessary
+	var experimentConfiguration map[string]string
+	experimentConfiguration["min_thread_count"] = strconv.Itoa(minThreadCount)
+	experimentConfiguration["max_thread_count"] = strconv.Itoa(maxThreadCount)
+	experimentConfiguration["load_points"] = loadDuration.String()
+	// Saving command line arguments and name of the binary might help with future analysis.
+	experimentConfiguration["command_arguments"] = strings.Join(os.Args[1:], " ")
```

Comment:
@ppalucki - I think it might be valuable to an experimenter. Shows what flags were quickly overridden in particular run,